jgiven-core/src/main/java/com/tngtech/jgiven/impl/ScenarioModelBuilder.java

```diff
@@ -172,6 +174,9 @@ public void stepMethodInvoked( Method method, List<NamedArgument> arguments, Inv
         } else if( method.isAnnotationPresent( StepComment.class ) ) {
             stepCommentAdded( arguments );
         } else {
+            addTags( method.getAnnotations() );
+            addTags( method.getDeclaringClass().getAnnotations() );
```

Comment:
I tried passing the receiver from cglib instead of using `getDeclaringClass`, but cglib subclasses the class to create the proxy which means any annotation not tagged with `@Inherited` gets lost. I'm not really sure if either way would be better than the other, so maybe this is completely fine…?